apps/dashboard/pkg/migration/schemaversion/v27.go

```diff
@@ -0,0 +1,172 @@
+package schemaversion
+
+// V27 migrates repeated panels and constant variables.
+//
+// The migration performs two main tasks:
+// 1. Removes repeated panel leftovers by filtering out panels with repeatPanelId or repeatByRow
+// 2. Migrates constant variables to textbox variables with proper current/options structure
+//
+// The migration includes comprehensive logic from the frontend:
+// - Panel filtering to remove repeated panels
+// - Constant variable migration with proper current/options structure
+// - Support for both constant and textbox variable types
+//
+// Example before migration:
+//
+//	"panels": [
+//	  {
+//	    "id": 1,
+//	    "type": "graph",
+//	    "repeatPanelId": "panel1"
+//	  },
+//	  {
+//	    "id": 2,
+//	    "type": "row",
+//	    "panels": [
+//	      {
+//	        "id": 3,
+//	        "type": "graph",
+//	        "repeatPanelId": "panel2"
+//	      }
+//	    ]
+//	  }
+//	],
+//	"templating": {
+//	  "list": [
+//	    {
+//	      "name": "constant_var",
+//	      "type": "constant",
+//	      "query": "default_value",
+//	      "hide": 0
+//	    }
+//	  ]
+//	}
+//
+// Example after migration:
+//
+//	"panels": [
+//	  {
+//	    "id": 2,
+//	    "type": "row",
+//	    "panels": []
+//	  }
+//	],
+//	"templating": {
+//	  "list": [
+//	    {
+//	      "name": "constant_var",
+//	      "type": "textbox",
+//	      "query": "default_value",
+//	      "current": {
+//	        "selected": true,
+//	        "text": "default_value",
+//	        "value": "default_value"
+//	      },
+//	      "options": [
+//	        {
+//	          "selected": true,
+//	          "text": "default_value",
+//	          "value": "default_value"
+//	        }
+//	      ],
+//	      "hide": 0
+//	    }
+//	  ]
+//	}
+func V27(dashboard map[string]interface{}) error {
+	dashboard["schemaVersion"] = 27
+
+	// Remove repeated panels
+	if panels, ok := dashboard["panels"].([]interface{}); ok {
+		dashboard["panels"] = removeRepeatedPanels(panels)
+	}
+
+	// Migrate constant variables
+	if templating, ok := dashboard["templating"].(map[string]interface{}); ok {
+		if list, ok := templating["list"].([]interface{}); ok {
+			for _, v := range list {
+				if variable, ok := v.(map[string]interface{}); ok {
+					migrateConstantVariable(variable)
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// removeRepeatedPanels filters out panels with repeatPanelId or repeatByRow properties
+// and cleans up repeated panels in collapsed rows
+func removeRepeatedPanels(panels []interface{}) []interface{} {
+	newPanels := make([]interface{}, 0, len(panels))
```

Comment:
since we do not know the exact length beforehand, we shouldn't preallocate
```suggestion
	newPanels := []interface{}{}
```